### PR TITLE
fuzz: addrman, avoid `ConsumeDeserializable` when possible

### PR DESCRIPTION
**Pull request by brunoerg**
Using specific functions like `ConsumeService`, `ConsumeAddress` and `ConsumeNetAddr` may be more effective than using `ConsumeDeserializable`. They always return some value while `ConsumeDeserializable` may return `std::nullopt`.

E.g.: In this part of the code, if `op_net_addr` is `std::nullopt`,  we basically generated the addresses (if so) unnecessarily, because we won't be able to use them:
```cpp
std::vector<CAddress> addresses;
LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 10000) {
    const std::optional<CAddress> opt_address = ConsumeDeserializable<CAddress>(fuzzed_data_provider);
    if (!opt_address) {
        break;
    }
    addresses.push_back(*opt_address);
}
const std::optional<CNetAddr> opt_net_addr = ConsumeDeserializable<CNetAddr>(fuzzed_data_provider);
if (opt_net_addr) {
    addr_man.Add(addresses, *opt_net_addr, std::chrono::seconds{ConsumeTime(fuzzed_data_provider, 0, 100000000)});
}
```

Also, if we are not calling `Add` effectively, it would also be affect other functions that may "depend" on it.

### CODE REVIEW

**Review by dergoegge — APPROVED** [review]
Code review ACK 025fda0a76893d09d19ec9a6c0ba86ad11c466f7

File: src/test/fuzz/addrman.cpp [hunk 1]
```diff
@@ -326,4 +307,4 @@ FUZZ_TARGET_INIT(addrman_serdeser, initialize_addrman)
     data_stream << addr_man1;
     data_stream >> addr_man2;
     assert(addr_man1 == addr_man2);
-}
+}
```
Comment (fanquake) [inline src/test/fuzz/addrman.cpp:310]: It'd be good to avoid making unrelated changes like this (deleting the file ending newline).
